src/api/app/models/kiwi/image/xml_parser.rb

```diff
@@ -103,6 +106,23 @@ def description
           specification:    attributes['specification']
         )
       end
+
+      def preference_type
+        # Do not support kiwi files where multiple type blocks are defined
+        return unless preference_type_attributes.is_a? Hash
+
+        Kiwi::PreferenceType.new(
```

Comment:
[nitpick] I see that we already did that in the other methods as well but I don't like that we hardcode the class name here. The parser does not need to know the class name, we could just do

``
@image.build_preference_type(...)
``